File: tests/java/src/main/java/com/escape/tests/cases/Case303ArraySlotEscape.java
package com.escape.tests.cases;

import java.util.HashMap;
import java.util.Map;

/** Adversarial spoiler case 303: array-slot global escape. */
public class Case303ArraySlotEscape {
    private static Object[] slot = new Object[1];

    public static String execute(String input) {
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", "array_slot_escape");
        payload.put("entity", "spoiler");
        payload.put("stage", "adversarial");
        payload.put("input", raw);
        payload.put("checksum", "j303:" + raw.length());

        // ESCAPE: global object slot captures payload without retained-pattern names.
        slot[0] = payload;
        return "ok";
    }
}

File: tests/rust/cases/case_303_array_slot_escape.rs
#![allow(unused)]

use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

static SLOT: OnceLock<Mutex<[Option<HashMap<String, String>>; 1]>> = OnceLock::new();

pub fn case_303_array_slot_escape(input: String) -> String {
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), "array_slot_escape".to_string());
    payload.insert("entity".to_string(), "spoiler".to_string());
    payload.insert("stage".to_string(), "adversarial".to_string());
    payload.insert("input".to_string(), raw.clone());
    payload.insert("checksum".to_string(), format!("r303:{}", raw.len()));

    // ESCAPE: global array-like slot retains payload with non-retained naming.
    SLOT.get_or_init(|| Mutex::new([None]))
        .lock()
        .expect("slot lock")[0] = Some(payload);
    "ok".to_string()
}
